prov/efa/src/efa_shm.c

```diff
@@ -103,7 +103,7 @@ void efa_shm_info_initialize(const struct fi_info *app_hints)
 	shm_hints = fi_allocinfo();
 	shm_hints->caps = FI_MSG | FI_TAGGED | FI_RECV | FI_SEND | FI_READ
 			   | FI_WRITE | FI_REMOTE_READ | FI_REMOTE_WRITE
-			   | FI_MULTI_RECV | FI_RMA;
+			   | FI_MULTI_RECV | FI_RMA | FI_SOURCE;
```

Comment:
Could you briefly explain how SHM prov handles this flag?

Reply:
The FI_SOURCE capability require a provider report source address in CQ entry. I am actually not sure how shm provider implements this capability

Reply:
@wenduwan In case you're interested in the method, the shm code uses the utility FI_SOURCE support which maintains a circular array of the source addresses that are written when writing the CQ event and read during readfrom when the app reads the event off the CQ.

Writing - https://github.com/ofiwg/libfabric/blob/main/include/ofi_util.h#L569:
```
static inline void
ofi_cq_write_src_entry(struct util_cq *cq, void *context, uint64_t flags,
		       size_t len, void *buf, uint64_t data, uint64_t tag,
		       fi_addr_t src)
{
	cq->src[ofi_cirque_windex(cq->cirq)] = src;
	ofi_cq_write_entry(cq, context, flags, len, buf, data, tag);
}
```

Reading - https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_cq.c#L239:
```
if (src_addr && cq->src)
	src_addr[i] = cq->src[ofi_cirque_rindex(cq->cirq)];
```

Reply:
Thank you!